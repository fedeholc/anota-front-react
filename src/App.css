.app__container {
  display: grid;
  grid-template-rows: auto 1fr auto;
  min-height: 100vh;
  margin: 0;
}

.app_main-grid {
  display: grid;
  grid-template-rows: "1fr";
}

.app__sticky-navbar {
  position: sticky;
  z-index: 99;
  top: 0;
  padding: 10px;
  background-color: hsl(0, 0%, 99%);
  border-bottom: 1px solid hsl(0, 0%, 50%);
  opacity: 95%;
  font-family: "Roboto Serif", serif;
  font-size: 1.2rem;
  font-weight: 700;

}

.top-toolbar {
  position: sticky;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  flex-wrap: nowrap;

  gap: 1rem;
  z-index: 99;
  top: 0;
  padding: 10px;
  background-color: hsl(0, 0%, 99%);
  border-bottom: 1px solid hsl(0, 0%, 50%);
  box-shadow: gray 0px 0px 3px 0px;


}

.top-toolbar__search {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  gap: 1rem;

}

.top-toolbar__new {
  margin-right: 1rem;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  gap: 1rem;
}

@media only screen and (max-width: 550px) {


  /* hace que se ajuste el ancho para que
  la toolbar quede centrada */
  .top-toolbar__search {
    width: min-content;
  }

  .app__sticky-navbar {
    display: none;
  }

}