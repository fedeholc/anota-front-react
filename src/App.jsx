import "./App.css";
/* import { useEffect, useState } from "react";
import { v4 as uuidv4 } from "uuid";
import ContentEditable from "react-contenteditable"; */
import { NotesProvider } from "./NotesContext";
import NotesList from "./NotesList";
import NotesInput from "./NoteInput";

//const API_URL = "https://anotes1-production.up.railway.app";
const API_URL = "http://localhost:3025";

function App() {
  return (
    <>
      <h1>provider:</h1>
      <NotesProvider>
        <NotesInput></NotesInput>
        <NotesList></NotesList>
      </NotesProvider>
    </>
  );
}

export default App;
