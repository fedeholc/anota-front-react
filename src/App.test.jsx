import { render, screen } from "@testing-library/react";
import { describe, it, expect } from "vitest";

import App from "./App";

describe("App", () => {
  it("renders headline", () => {
    render(<App />);
    screen.debug();
    expect(screen.getByText(/anota/i)).toBeInTheDocument();
  });
  it("title snapshot", () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
