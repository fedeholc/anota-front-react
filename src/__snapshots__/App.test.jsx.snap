// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > title snapshot 1`] = `
<div>
  <div
    class="app__container"
  >
    <div
      class="app__main-grid"
    >
      <div
        class="_auth__info_65f7c7"
      />
      <div
        class="top-toolbar-wrapper"
      >
        <div
          class="top-nombre"
        >
          📓ANOTA
        </div>
        <div
          class="top-toolbar"
        >
          <div
            class="top-toolbar__search"
          >
            <div>
              <span
                class="ant-input-group-wrapper ant-input-group-wrapper-outlined css-dev-only-do-not-override-1k979oh ant-input-search"
                style="width: 200px;"
              >
                <span
                  class="ant-input-wrapper ant-input-group css-dev-only-do-not-override-1k979oh"
                >
                  <span
                    class="ant-input-affix-wrapper css-dev-only-do-not-override-1k979oh ant-input-outlined"
                  >
                    <input
                      class="ant-input css-dev-only-do-not-override-1k979oh"
                      placeholder="Buscar..."
                      type="text"
                      value=""
                    />
                    <span
                      class="ant-input-suffix"
                    >
                      <span
                        class="ant-input-clear-icon ant-input-clear-icon-hidden"
                        role="button"
                        tabindex="-1"
                      >
                        <span
                          aria-label="close-circle"
                          class="anticon anticon-close-circle"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="close-circle"
                            fill="currentColor"
                            fill-rule="evenodd"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                            />
                          </svg>
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="ant-input-group-addon"
                  >
                    <button
                      class="ant-btn css-dev-only-do-not-override-1k979oh ant-btn-default ant-btn-icon-only ant-input-search-button"
                      type="button"
                    >
                      <span
                        class="ant-btn-icon"
                      >
                        <span
                          aria-label="search"
                          class="anticon anticon-search"
                          role="img"
                        >
                          <svg
                            aria-hidden="true"
                            data-icon="search"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="64 64 896 896"
                            width="1em"
                          >
                            <path
                              d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                            />
                          </svg>
                        </span>
                      </span>
                    </button>
                  </span>
                </span>
              </span>
            </div>
            <div>
              <div
                class="ant-select ant-select-outlined css-dev-only-do-not-override-1k979oh ant-select-multiple ant-select-allow-clear ant-select-show-arrow ant-select-show-search"
                style="width: 200px;"
              >
                <div
                  class="ant-select-selector"
                >
                  <div
                    class="ant-select-selection-overflow"
                  >
                    <div
                      class="ant-select-selection-overflow-item ant-select-selection-overflow-item-suffix"
                      style="opacity: 1;"
                    >
                      <div
                        class="ant-select-selection-search"
                        style="width: 0px;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-controls="rc_select_TEST_OR_SSR_list"
                          aria-expanded="false"
                          aria-haspopup="listbox"
                          aria-owns="rc_select_TEST_OR_SSR_list"
                          autocomplete="off"
                          class="ant-select-selection-search-input"
                          id="rc_select_TEST_OR_SSR"
                          readonly=""
                          role="combobox"
                          style="opacity: 0;"
                          type="search"
                          unselectable="on"
                          value=""
                        />
                        <span
                          aria-hidden="true"
                          class="ant-select-selection-search-mirror"
                        >
                           
                        </span>
                      </div>
                    </div>
                  </div>
                  <span
                    class="ant-select-selection-placeholder"
                  >
                    Buscar por tag...
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="top-toolbar__new"
          >
            <div>
              <span
                aria-label="right"
                class="anticon anticon-right toolbar__icon"
                role="img"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </div>
            <div>
              <div>
                <svg
                  class="toolbar__icon"
                  fill="none"
                  stroke="#000000"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                    stroke="none"
                  />
                  <path
                    d="M4 4m0 2a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-12a2 2 0 0 1 -2 -2z"
                  />
                  <path
                    d="M4 14m0 2a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2v2a2 2 0 0 1 -2 2h-12a2 2 0 0 1 -2 -2z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="toolbar__button-container"
            >
              <span
                aria-label="plus"
                class="anticon anticon-plus toolbar__icon"
                role="img"
                tabindex="-1"
              >
                <svg
                  aria-hidden="true"
                  data-icon="plus"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                  />
                  <path
                    d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
                  />
                </svg>
              </span>
            </div>
          </div>
        </div>
      </div>
      <div>
        <div>
          <div
            style="display: grid; place-content: center; margin-top: 3rem;"
          >
            <button>
              Entrar como invitado
            </button>
          </div>
          <div
            class="_auth__wrapper_65f7c7"
          >
            <div
              class="_auth__container_65f7c7"
            >
              <div
                class="_auth_msg_65f7c7"
              >
                Por favor ingrese su email y contraseña, o regístrese si aún no lo hizo.
              </div>
              <div
                class="supa_auth"
              >
                <div
                  class=""
                >
                  <div
                    class="supabase-auth-ui_ui-container c-jTXIoq c-jTXIoq-bsgKCL-direction-vertical c-jTXIoq-bXHFxK-gap-large"
                    direction="vertical"
                    gap="large"
                  >
                    <div
                      class="supabase-auth-ui_ui-container c-jTXIoq c-jTXIoq-bsgKCL-direction-vertical c-jTXIoq-jjTuOt-gap-small"
                      direction="vertical"
                      gap="small"
                    >
                      <button
                        class="supabase-auth-ui_ui-button c-bOcPnF c-bOcPnF-iwjZXY-color-default"
                      >
                        <svg
                          class="c-eSSyNk"
                          height="21px"
                          viewBox="0 0 48 48"
                          width="21px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M43.611,20.083H42V20H24v8h11.303c-1.649,4.657-6.08,8-11.303,8c-6.627,0-12-5.373-12-12c0-6.627,5.373-12,12-12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C12.955,4,4,12.955,4,24c0,11.045,8.955,20,20,20c11.045,0,20-8.955,20-20C44,22.659,43.862,21.35,43.611,20.083z"
                            fill="#FFC107"
                          />
                          <path
                            d="M6.306,14.691l6.571,4.819C14.655,15.108,18.961,12,24,12c3.059,0,5.842,1.154,7.961,3.039l5.657-5.657C34.046,6.053,29.268,4,24,4C16.318,4,9.656,8.337,6.306,14.691z"
                            fill="#FF3D00"
                          />
                          <path
                            d="M24,44c5.166,0,9.86-1.977,13.409-5.192l-6.19-5.238C29.211,35.091,26.715,36,24,36c-5.202,0-9.619-3.317-11.283-7.946l-6.522,5.025C9.505,39.556,16.227,44,24,44z"
                            fill="#4CAF50"
                          />
                          <path
                            d="M43.611,20.083H42V20H24v8h11.303c-0.792,2.237-2.231,4.166-4.087,5.571c0.001-0.001,0.002-0.001,0.003-0.002l6.19,5.238C36.971,39.205,44,34,44,24C44,22.659,43.862,21.35,43.611,20.083z"
                            fill="#1976D2"
                          />
                        </svg>
                        Sign in with Google
                      </button>
                      <button
                        class="supabase-auth-ui_ui-button c-bOcPnF c-bOcPnF-iwjZXY-color-default"
                      >
                        <svg
                          class="c-eSSyNk"
                          fill="gray"
                          height="21px"
                          viewBox="0 0 30 30"
                          width="21px"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                           
                          <path
                            d="M15,3C8.373,3,3,8.373,3,15c0,5.623,3.872,10.328,9.092,11.63C12.036,26.468,12,26.28,12,26.047v-2.051 c-0.487,0-1.303,0-1.508,0c-0.821,0-1.551-0.353-1.905-1.009c-0.393-0.729-0.461-1.844-1.435-2.526 c-0.289-0.227-0.069-0.486,0.264-0.451c0.615,0.174,1.125,0.596,1.605,1.222c0.478,0.627,0.703,0.769,1.596,0.769 c0.433,0,1.081-0.025,1.691-0.121c0.328-0.833,0.895-1.6,1.588-1.962c-3.996-0.411-5.903-2.399-5.903-5.098 c0-1.162,0.495-2.286,1.336-3.233C9.053,10.647,8.706,8.73,9.435,8c1.798,0,2.885,1.166,3.146,1.481C13.477,9.174,14.461,9,15.495,9 c1.036,0,2.024,0.174,2.922,0.483C18.675,9.17,19.763,8,21.565,8c0.732,0.731,0.381,2.656,0.102,3.594 c0.836,0.945,1.328,2.066,1.328,3.226c0,2.697-1.904,4.684-5.894,5.097C18.199,20.49,19,22.1,19,23.313v2.734 c0,0.104-0.023,0.179-0.035,0.268C23.641,24.676,27,20.236,27,15C27,8.373,21.627,3,15,3z"
                          />
                        </svg>
                        Sign in with Github
                      </button>
                    </div>
                  </div>
                  <div
                    class="supabase-auth-ui_ui-divider c-kbVGyA"
                  />
                  <form
                    autocomplete="on"
                    id="auth-sign-in"
                    style="width: 100%;"
                  >
                    <div
                      class="supabase-auth-ui_ui-container c-jTXIoq c-jTXIoq-bsgKCL-direction-vertical c-jTXIoq-bXHFxK-gap-large"
                      direction="vertical"
                      gap="large"
                    >
                      <div
                        class="supabase-auth-ui_ui-container c-jTXIoq c-jTXIoq-bsgKCL-direction-vertical c-jTXIoq-bXHFxK-gap-large"
                        direction="vertical"
                        gap="large"
                      >
                        <div>
                          <label
                            class="supabase-auth-ui_ui-label c-bpexlo"
                            for="email"
                          >
                            Email address
                          </label>
                          <input
                            autocomplete="email"
                            class="supabase-auth-ui_ui-input c-dEnagJ c-dEnagJ-bBzSYw-type-default"
                            id="email"
                            name="email"
                            placeholder="Your email address"
                            type="email"
                            value=""
                          />
                        </div>
                        <div>
                          <label
                            class="supabase-auth-ui_ui-label c-bpexlo"
                            for="password"
                          >
                            Your Password
                          </label>
                          <input
                            autocomplete="current-password"
                            class="supabase-auth-ui_ui-input c-dEnagJ c-dEnagJ-bBzSYw-type-password"
                            id="password"
                            name="password"
                            placeholder="Your password"
                            type="password"
                            value=""
                          />
                        </div>
                      </div>
                      <button
                        class="supabase-auth-ui_ui-button c-bOcPnF c-bOcPnF-cmFMMs-color-primary"
                        type="submit"
                      >
                        Sign in
                      </button>
                      <div
                        class="supabase-auth-ui_ui-container c-jTXIoq c-jTXIoq-bsgKCL-direction-vertical c-jTXIoq-jjTuOt-gap-small"
                        direction="vertical"
                        gap="small"
                      >
                        <a
                          class="supabase-auth-ui_ui-anchor c-dumjqv"
                          href="#auth-forgot-password"
                        >
                          Forgot your password?
                        </a>
                        <a
                          class="supabase-auth-ui_ui-anchor c-dumjqv"
                          href="#auth-sign-up"
                        >
                          Don't have an account? Sign up
                        </a>
                      </div>
                    </div>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="padding: 1rem; margin: auto;"
      />
    </div>
    <footer>
      Federico Holc ⚡️ 2024
    </footer>
  </div>
</div>
`;
